packages/rest-client-jsonapi/src/services/utils/fetch/fetch.ts

```diff
@@ -17,8 +18,10 @@ export async function fetchJsonApi<T>(
     method,
     body: body ? JSON.stringify({ data: body }) : undefined,
     headers: {
-      "Content-Type": "application/vnd.api+json",
+      ...fetchOptions?.headers,
+      "Content-Type": "application/vnd.api+json", // This will override any Content-Type in fetchOptions
```

Comment:
Howdy. 

I kept that header order under the assumption that changing the content-type would cause issues. Now, having tested other content-types I see that the request content type can be changed but the response is still going to be json and I don't see any issues (albeit in limited testing).

Otherwise, I don't really have an opinion about whether overriding should be an option.

Reply:
hey @kt-mattie-langenberg even if doesn't break, i think you're right on the original code and we shouldn't allow changing the content-type, since this lib is modeled/following json api:

> JSON:API requires use of the JSON:API media type ([application/vnd.api+json](http://www.iana.org/assignments/media-types/application/vnd.api+json)) for exchanging data.

https://jsonapi.org/format/

Reply:
i tried hacking together some type to make it so we use a custom request init that doesn't allow content-type but it seems like a major pain... ie

```
type Example = Omit<RequestInit, 'headers'> & {
  headers?: HeadersInit <---
};
```

the problem is `HeadersInit` accepts `[string, string][]` and `Record<string, string>`, so not sure off the top of my head how to write the type for not allowing `[['Content-Type', 'asdfadfa']]`. if it's not easy to figure out, leave as is (don't allow overwriting content-type)